docs/layer_util.js

```diff
@@ -80,69 +113,65 @@ function setMapToDrawLayersOn(map) {
 /**
  * Toggles on displaying an asset on the map.
  *
- * @param {string} layerName
+ * @param {Object} layer Data for layer coming from Firestore
  * @param {google.maps.Map} map main map
  */
-function toggleLayerOn(layerName, map) {
-  const currentLayerMapValue = layerMap.get(layerName);
-  currentLayerMapValue.displayed = true;
-  if (currentLayerMapValue.data) {
-    addLayerFromFeatures(currentLayerMapValue, layerName);
+function toggleLayerOn(layer, map) {
+  const index = layer['index'];
+  const layerMapValue = layerArray[index];
+  layerMapValue.displayed = true;
+  if (layerMapValue.data) {
```

Comment:
Is this still the right check for image layers? Am I understanding correctly that those have an overlay attr instead of a data attr? Do we not cache image layers?

Reply:
Very nice catch! I realized this last night after mailing for review, but it's not a new bug in this PR, so I was going to fix it in a follow-up. Added a TODO.